src/store_client.cc

```diff
@@ -176,14 +176,12 @@ store_client::finishCallback()
 
     cmp_offset = result.offset + result.length;
     STCB *temphandler = _callback.callback_handler;
-    void *cbdata = _callback.callback_data;
+    const auto cbdata = _callback.cbData.validDone();
     _callback = Callback(nullptr, nullptr);
```

Comment:
Please also fix this line.
```suggestion
    _callback = Callback();
```

That may expose the issues with C++11 initialization of function pointers we have hit elsewhere. Namely that the constructor needs to set them explicitly, not implicitly.